cpu/atmega_common/periph/i2c.c

```diff
@@ -251,12 +270,14 @@ int i2c_write_regs(i2c_t dev, uint8_t address, uint8_t reg, const void *data, in
 void i2c_poweron(i2c_t dev)
 {
     assert(dev < I2C_NUMOF);
+    (void) dev;
```

Comment:
You could enclose the `(void) dev;` with `#ifdef NDEBUG`

Reply:
IMHO this will decrease readability and the compiler with optimise out one or the other anyway.

Reply:
I agree with @smlng. This would lead to very ugly code.